modules/renter.go

```diff
@@ -50,12 +55,154 @@ type Allowance struct {
 // DownloadInfo provides information about a file that has been requested for
 // download.
 type DownloadInfo struct {
-	SiaPath     string    `json:"siapath"`
-	Destination string    `json:"destination"`
-	Filesize    uint64    `json:"filesize"`
-	Received    uint64    `json:"received"`
-	StartTime   time.Time `json:"starttime"`
-	Error       string    `json:"error"`
+	SiaPath     string         `json:"siapath"`
+	Destination DownloadWriter `json:"destination"`
+	Filesize    uint64         `json:"filesize"`
+	Received    uint64         `json:"received"`
+	StartTime   time.Time      `json:"starttime"`
+	Error       string         `json:"error"`
+}
+
+// DownloadWriter provides an interface which all output writers have to implement.
+type DownloadWriter interface {
+	WriteAt(b []byte, off int64) (int, error)
+	Location() string
```

Comment:
The download logic in siac relies on the `downloads` endpoint providing a location parameter that matches the destination provided in the download command. Thus, I had to keep this information around. Since the `DownloadWriter` is the destination for all data, it seemed fitting to use it as the place that provides this information instead. 

Having it in DownloadWriter also means that the DownloadWriter implementation defines what the value is (if it is hard-coded as is the case with the Http writer). If it is stored in the struct, this information would somehow have to be supplied when the struct is created.